wikipedia_cleanup/data_processing.py

```diff
@@ -56,45 +64,126 @@ def read_pickle_file(file_path: Path) -> List[InfoboxChange]:
+    :param n_files :Optional[int]:
+    Number of files to read from the input Folder. None means using all Files.
+    :param n_jobs: int:
+    Number of Jobs / Processes used for parallel reads.
+    :param filters: Optional[List[AbstractDataFilter]]:
+    (Ordered) List of Filters
+    that should be applied on the fly when loading.
+    Consider using: `get_default_filters()`
+    :return: Tuple of: All change items from all read files
+    where each all files are sorted after
+    (page_id, infobox_key, property_name, value_valid_from),
+    the resulting filters hold the accumulated stats of the read.
+    """
+    if filters is None:
+        filters = []
     files = [x for x in Path(input_path).rglob("*.output.json") if x.is_file()]
     files.extend([x for x in Path(input_path).rglob("*.pickle") if x.is_file()])
     if n_files is not None:
         n_jobs = min(n_jobs, n_files)
     n_jobs = min(n_jobs, len(files))
     files = files[slice(n_files)]
     if n_jobs > 1:
-        all_changes = process_map(read_file, files, max_workers=n_jobs)
+        all_changes, mapped_filters = zip(
+            *process_map(
+                read_and_filter_file,
+                zip(files, [filters] * len(files)),
+                max_workers=n_jobs,
+            )
+        )
     else:
         all_changes = []
-        for file in tqdm(files):
-            all_changes.append(read_file(file))
+        mapped_filters = []
+        for file, data_filters in tqdm(
+            zip(files, [deepcopy(filters) for _ in range(len(files))])
+        ):
+            changes_and_filters = read_and_filter_file((file, data_filters))
+            all_changes.append(changes_and_filters[0])
+            mapped_filters.append(changes_and_filters[1])
     all_changes = itertools.chain.from_iterable(all_changes)
-    return pd.DataFrame([change.__dict__ for change in all_changes])
+    filters = merge_filter_stats(mapped_filters)
+    return pd.DataFrame([change.__dict__ for change in all_changes]), filters
 
 
-# test
+# local test
 if __name__ == "__main__":
-    get_data(
+    _, merged_filters = get_data(
         Path("/home/secret/uni/Masterprojekt/data/test_case_data/output-infobox"),
         1000,
         3,
     )
-    get_data(
+    assert len(merged_filters) == 0
+    _, merged_filters = get_data(
         Path(
             "/run/media/secret/manjaro-home/secret/mp-data/"
             "costum-format-filtered-dayly/costum-format-filtered-dayly"
         ),
         5,
         3,
+        filters=generate_default_filters(),
+    )
+    print(get_stats_from_filters(merged_filters))
+    _, merged_filters = get_data(
+        Path(
+            "/run/media/secret/manjaro-home/secret/mp-data/"
+            "costum-format-filtered-dayly/costum-format-filtered-dayly"
+        ),
+        5,
+        1,
+        filters=generate_default_filters(),
     )
+    print(get_stats_from_filters(merged_filters))
```

Comment:
remember to remove later

Reply:
What exactly do you mean? I would not remove it in this PR but at a later point. E.g. refactoring part at the end of the project.

Reply:
thats exactly what i mean, remember to remove this at a later point